test_framework/key.py

```diff
@@ -588,12 +611,29 @@ def tweak_add(self, tweak):
         ret.set(tweaked.to_bytes(32, 'big'), self.compressed)
         return ret
 
-def generate_schnorr_nonce():
-    """Generate a random valid bip-schnorr nonce.
+def generate_key_pair(secret=None, compressed=True):
+    """Convenience function to generate a private-public key pair."""
+    d = ECKey()
+    if secret:
+        d.set(secret, compressed)
+    else:
+        d.generate(compressed)
+
+    P = d.get_pubkey()
+    return d, P
+
+def generate_schnorr_nonce(seed=None):
```

Comment:
What do you expect to be passed here?
Because the deterministic nonce should be over the bytes of private key and the bytes of the hashed message.
the only way to use this function correctly is like this `generate_schnorr_nonce(d.to_bytes() + e.to_bytes())`

Reply:
Sorry - this wasn't clear. The reason I want to do this is for testing purposes. I want to be able to set the nonce to a known value so we can assert that the student are doing the correct thing in the notebook. If that's confusing, I'll just drop this.

Reply:
removed this functionality